### ISSUE TITLE
Design TWM - tabsContainer_v0.0.1

### ISSUE THREAD

**Issue by Mehrabbruno** [wizardamigos/wizardamigos.github.io#114]
## `@todo`

https://github.com/wizardamigos/wizardamigos.github.io/issues/111

  - [x] Design - tabsContiainer_v0.0.1
    - `@input` 📦 [WF 14: TWM specs in WF -Iteration 2- figma file`](https://www.figma.com/file/lk41O0wlkMEdPiCBGVMOO8/?node-id=897%3A11280) from #101 
    - `@input` :package: [`twm ui/ux specification`](https://hackmd.io/yrqMJjhaQm2b8MhS4Gqxhg?view) from #111 

    - 
     - [ ] Tab Container Component (contains)
        - [ ] Tabs from #112 
        - [ ] Collapsed tabs counter CTA
           - [ ] Collapsed tabs pop-over
              - [ ] pop-over header (fixed to the top and contains title, total number of tabs and close button)
              - [ ] Collapsed tabs (contains program, title, and close button)
              - [ ] Search input (contains search icon, inputField and close icon when added text)
        - [ ] Action List CTA
        - [ ] Add tabs CTA

    - `@output` 🏭 [`TWM iteration 15 Figma File`](https://www.figma.com/file/PiQmPX3q1V2Ph7hARR2jOm/tiling-window-manager-3?node-id=224%3A2927) from [`comment`]()

## `@info`

for [`TWM iteration 15 Figma File`](https://www.figma.com/file/PiQmPX3q1V2Ph7hARR2jOm/tiling-window-manager-3?node-id=0%3A1) we need to design the tab containers. These will be house the tab components. read more on `twm ui/ux specification`

**Comment by Mehrabbruno** [comment]
* `done` Design - tabsContiainer_v0.0.1 
   * `done` Tab Container Component (contains) 
      * `done` #112 
      * `done` Collapsed tabs counter CTA 
         * `done` #121 
      * `done` Action List CTA
      * `done` Add tabs CTA
      * `@output` 📦 [`tabContainer_v0.0.1`](https://www.figma.com/file/PiQmPX3q1V2Ph7hARR2jOm/tiling-window-manager-3?node-id=318%3A2739)
   * `done` Record an explanatory video for Tab Container Component
      * `@output`  📦 [`video recording`](https://youtu.be/hzvaO3L_4kY)

**Comment by Mehrabbruno** [comment]
* `done` Set the border radius of the tabContianer_v0.0.1 to 0px 
   * `@output` [`tabContainer_v0.0.2`](https://www.figma.com/file/PiQmPX3q1V2Ph7hARR2jOm/tiling-window-manager-3?node-id=318%3A2738)

**Comment by Mehrabbruno** [comment]
* `done` Create large button
   * Create a button with 14px x 81px padding
   * Add border-radius of 10px
   * set font-size to 24px
   * `@output` [`lg-button_v0.0.1`](https://www.figma.com/file/PiQmPX3q1V2Ph7hARR2jOm/tiling-window-manager-3?node-id=318%3A2738)

**Comment by serapath** [comment]
**feedback `2022.10.14`**

i noticed one thing that we might need to improve, so if you are now learning and going through the codecamp lessons, that is great

but let me show you on one example

![img](https://imgur.com/ewtIu2S.png)

david for example is currently working on the design of a website, but once he finishes i would let him go through your TWM issue and your videos to learn how we use figma and then adopt that style too, so we should really make things **"perfect"** for others to learn from that example



So, one rule that is not well presented in that screenshot (but it's similar to other issues), is that every `@input` should be a sub bullet point to exactly the `* [ ] example task do XYZ` that actually needs it as an input. In the example above this means something like:

```
* [x] design tabsContainer
  * [x] #112 (= design tab)
  * [x] import and integrate designed tab
    * `@input` :package: [`tab_v0.0.1`](#link_to_figma) from #112
  * [x] #121 (=design collapsedTabsPopover)
  * [x] add collapsed tab counter CTA for collapsedTabsPopover
    * `@input` :package: [`collapsedTabsPopover_v0.0.1`](#link_to_figma) from #121
  * [x] add action list CTA
  * [x] add tabs CTA
  * [x] submit worklog
  * `@output` :package: [`tabContainer_v0.0.1`](#link_to_figma) from [`comment`](#link_to_comment)
    * next #111
```

so, the `record explanatory video` i assume is the worklog and generally all the project management overhead.
This is the part that you can add as a textual todo in terms of `submit worklog` (see above) to capture the time it takes to record the video, post the comment, update your timelog hackmd, etc..
But it is a generic activity for every task, so that is why this should not create an output. The screencast worklogs are the only type of output that only exist in worklog comments and get linked, because it is part of our general project management process and not part of the project TWM itself, but i agree a generic `submit worklog` todo that you can log time on should be included every time you do a worklog. that makes sense, so it can even appear multiple times to capture the effort.


the main different to the way you did it above is, that 
1. you have issue #121 and issue #112 linked as todos to actually craete the tab_v0.0.1 and collpsedTabsPopuver_v0.0.1, because they are needed so they need to be made first.
2. then once that is done, and  the output figmas for each of them exists as a file, we need to add todos to actuall integrate them into the desgin for the `tabsContainer`, which is
    * [ ] import and integrate design tab
    * [ ] add collapsed tab counter CTA

and then in the end you just have the output of the "design - tabContainer" task issue itself



every single issue should be structured like that instead of the way you did it.

**result:**
=> so less indentation
=> inputs are directly associated with the specific one or more  sub issues where they are actually needed

...but it is minor changes, you already did it mostly correct :slight_smile:

so could you adapt that?

**Comment by serapath** [comment]
**feedback `2022.10.22`**

![tabs container](https://imgur.com/krl30IL.png)

Here you mentioned clicking on the numbers (2) will show the remaining tabs popover so you can select one and i like it and the same goes for left and right side and those numbers only show up if the user opens many tabs.

But I think the hamburger menu was earlier described as having that function.
Do you think we can remove the hamburger menu, because indeed, the user can just click the numbers? That would be cool - one less menu or button to worry about :-)

**Comment by Mehrabbruno** [comment]
feedback `2022.10.24`

> But I think the hamburger menu was earlier described as having that function

The hamburger menu was intended for actions like split-screens. The counter for to toggle **_collapsedTabsPopover_** and hamburger menu to toggle _**`actionTabsPopover`**_

**Comment by Mehrabbruno** [comment]
feedback `2022.10.24`

> But I think the hamburger menu was earlier described as having that function

The hamburger menu was intended for actions like split-screens. The counter for to toggle **_collapsedTabsPopover_** and hamburger menu to toggle _**`actionTabsPopover`**_
